jerry-core/ecma/operations/ecma-objects-general.c

```diff
@@ -592,166 +592,162 @@ ecma_op_general_object_define_own_property (ecma_object_t *obj_p, /**< the objec
                                                                      property_desc_p->is_configurable);
 
+
+      JERRY_ASSERT (property_desc_p->is_value_defined
+                    || ecma_is_value_undefined (property_desc_p->value));
+
       ecma_named_data_property_assign_value (obj_p, new_prop_p, property_desc_p->value);
     }
     else
     {
       // b.
-      JERRY_ASSERT (is_property_desc_accessor_descriptor);
 
       ecma_create_named_accessor_property (obj_p,
                                            property_name_p,
                                            property_desc_p->get_p,
                                            property_desc_p->set_p,
                                            property_desc_p->is_enumerable,
                                            property_desc_p->is_configurable);
-
     }
 
     return ecma_make_simple_value (ECMA_SIMPLE_VALUE_TRUE);
   }
 
-  // 5.
-  if (is_property_desc_generic_descriptor
-      && !property_desc_p->is_enumerable_defined
-      && !property_desc_p->is_configurable_defined)
-  {
-    return ecma_make_simple_value (ECMA_SIMPLE_VALUE_TRUE);
-  }
-
   // 6.
-  ecma_property_types_t current_type = ECMA_PROPERTY_GET_TYPE (current_p);
-  const bool is_current_data_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDDATA);
-  const bool is_current_accessor_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDACCESSOR);
-
-  JERRY_ASSERT (is_current_data_descriptor || is_current_accessor_descriptor);
-
-  bool is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = true;
-  if (property_desc_p->is_value_defined)
-  {
-    if (!is_current_data_descriptor
-        || !ecma_op_same_value (property_desc_p->value,
-                                ecma_get_named_data_property_value (current_p)))
-    {
-      is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = false;
-    }
-  }
-
-  if (property_desc_p->is_writable_defined)
-  {
-    if (!is_current_data_descriptor
-        || property_desc_p->is_writable != ecma_is_property_writable (current_p))
-    {
-      is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = false;
-    }
-  }
+  ecma_property_types_t current_property_type = ECMA_PROPERTY_GET_TYPE (current_p);
+  const bool is_current_configurable = ecma_is_property_configurable (current_p);
 
-  if (property_desc_p->is_get_defined)
-  {
-    if (!is_current_accessor_descriptor
-        || property_desc_p->get_p != ecma_get_named_accessor_property_getter (current_p))
-    {
-      is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = false;
-    }
-  }
+  JERRY_ASSERT (current_property_type == ECMA_PROPERTY_TYPE_NAMEDDATA
+                || current_property_type == ECMA_PROPERTY_TYPE_NAMEDACCESSOR);
 
-  if (property_desc_p->is_set_defined)
+  /* If current is configurable, any change is allowed. */
+  if (!is_current_configurable)
```

Comment:
I do, especially here.

As an example, in step 11. a., we also have a 4-line condition which is intuitive to read: "if we want to overwrite a getter with something that's not the same as we already have OR we want to overwrite a setter with something that's not the same as we already have..."

Here, a very similar condition is broken to nested ifs for no apparent reason and makes me spend brain cycles on figuring out what needs special handling. Outcome: nothing. Our condition here is: "if current property is non-configurable but we either want to make it configurable or change its enumerable status..."

There is no else branch either in the inner or in the outer if. So, I see no point in nesting them. (Or do we have a guidline that predicates have to be in disjunctive normal form?)